File: implementations/flutter/design_patterns_flutter/lib/patterns/structural/bridge/order_report.dart
import 'abstraction.dart';
import 'domain/report_data.dart';

class Order {
  final String id;
  final String customer;
  final double total;
  final String status;
  const Order({required this.id, required this.customer, required this.total, required this.status});
}

class OrdersReport extends Report {
  final List<Order> orders;
  OrdersReport(super.exporter, this.orders);

  @override
  Future<ReportData> build() async {
    final rows = orders
        .map((o) => {
              'id': o.id,
              'customer': o.customer,
              'total': o.total,
              'status': o.status,
            })
        .toList();
    return ReportData(title: 'Orders Report', rows: rows);
  }
}

File: implementations/react-native/design-patterns-rn/src/patterns/structural/bridge/order-report.ts
import { Report } from './abstraction';
import type { Exporter } from './implementor';
import type { Order } from './interfaces/order';
import type { ReportData } from './interfaces/report-data';

export class OrdersReport extends Report {
  private orders: Order[];
  constructor(exporter: Exporter, orders: Order[]) { super(exporter); this.orders = orders; }

  async build(): Promise<ReportData> {
    const rows = this.orders.map((o) => ({ id: o.id, customer: o.customer, total: o.total, status: o.status }));
    return { title: 'Orders Report', rows };
  }
}
